fidimag/common/sundials/cvode.pyx

```diff
@@ -246,4 +301,194 @@ cdef class CvodeSolver(object):
         self.user_data.v = NULL
         self.user_data.jv = NULL
         N_VDestroy_Serial(self.u_y)
-        CVodeFree(&self.cvode_mem)
+        CVodeFree(& self.cvode_mem)
+
+cdef class CvodeSolver_OpenMP(object):
+    cdef public double t
+    cdef public np.ndarray y
+    cdef double rtol, atol
+    cdef int cvode_already_initialised
+    cdef np.ndarray y0
+    cdef np.ndarray dm_dt
+    cdef np.ndarray mp
+    cdef np.ndarray Jmp
+    cdef N_Vector u_y
+    cdef void * cvode_mem
+    cdef void * rhs_fun
+    cdef void * jvn_fun
+    cdef callback_fun
+    cdef jtimes_fun
+    cdef cv_userdata user_data
+    cdef long int nsteps, nfevals, njevals
+    cdef int max_num_steps
+    cdef int has_jtimes
+    cdef str linear_solver
+    cdef str parellel_solver
+    cdef int num_threads
+
+    def __cinit__(self, spins, rhs_fun, jtimes_fun=None, linear_solver="spgmr", rtol=1e-8, atol=1e-8):
+        if os.environ['OMP_NUM_THREADS']:
+            try:
+                self.num_threads = int(os.environ['OMP_NUM_THREADS'])
```

Comment:
I tried this, the problem is that because of the Global Interpreter Lock it always returns 1 when called from Python... There must be a way of doing it but I couldn't find it..

Sent from my iPhone

On 1 Jul 2016, at 13:05, Weiwei Wang <notifications@github.com<mailto:notifications@github.com>> wrote:


In fidimag/common/sundials/cvode.pyx<https://github.com/computationalmodelling/fidimag/pull/55#discussion_r69289293>:

> +    cdef void * rhs_fun
> +    cdef void * jvn_fun
> +    cdef callback_fun
> +    cdef jtimes_fun
> +    cdef cv_userdata user_data
> +    cdef long int nsteps, nfevals, njevals
> +    cdef int max_num_steps
> +    cdef int has_jtimes
> +    cdef str linear_solver
> +    cdef str parellel_solver
> +    cdef int num_threads
> +
> +    def __cinit__(self, spins, rhs_fun, jtimes_fun=None, linear_solver="spgmr", rtol=1e-8, atol=1e-8):
> +        if os.environ['OMP_NUM_THREADS']:
> +            try:
> +                self.num_threads = int(os.environ['OMP_NUM_THREADS'])


Alternatively, omp_get_max_threads() can be used to get the number of threads set by OMP_NUM_THREADS.

�
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/computationalmodelling/fidimag/pull/55/files/af758f955df4908580320ce0d453d2a27e06760a#r69289293>, or mute the thread<https://github.com/notifications/unsubscribe/AIJNUvGSDWSR86KhOqyxOdeZoIrX7UISks5qRQKCgaJpZM4JCrzH>.